Lib/http/server.py

```diff
@@ -1014,8 +1014,10 @@ def is_cgi(self):
         """
         collapsed_path = _url_collapse_path(self.path)
         dir_sep = collapsed_path.find('/', 1)
-        head, tail = collapsed_path[:dir_sep], collapsed_path[dir_sep+1:]
-        if head in self.cgi_directories:
+        while dir_sep > 0 and not collapsed_path[:dir_sep] in self.cgi_directories:
```

Comment:
The unit test should be done with appended or overridden `cgi_directories`

Reply:
Thank you so much for your kind explanation. I will upload my unit test code right away with the news and get back to you